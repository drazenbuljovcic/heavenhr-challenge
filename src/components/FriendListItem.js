import React, { Component } from 'react';
import { PropTypes } from 'prop-types';

import classnames from 'classnames';
import styles from './FriendListItem.css';

class FriendListItem extends Component {
  render() {
    return (
      <li className={styles.friendListItem}>
        <div className={styles.friendInfos}>
          <div>
            <span>{this.props.name}</span>
            <i className={classnames('fa', {
              'fa-male': this.props.gender === 'male',
              'fa-female': this.props.gender === 'female'
            })} aria-hidden="true" />
          </div>
          <div>
            <small>xx friends in common</small>
          </div>
        </div>
        <div className={styles.friendActions}>
          <button className={`btn btn-default ${styles.btnAction}`}
                  onClick={() => this.props.starFriend(this.props.id)}>
            <i className={classnames('fa', {
              'fa-star': this.props.starred,
              'fa-star-o': !this.props.starred
            })} />
          </button>
          <button className={`btn btn-default ${styles.btnAction}`}
                  onClick={() => this.props.deleteFriend(this.props.id)}>
            <i className="fa fa-trash" />
          </button>
        </div>
      </li>
    );
  }
}

FriendListItem.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  gender: PropTypes.string,
  starred: PropTypes.bool,
  starFriend: PropTypes.func.isRequired
};

FriendListItem.defaultProps = {
  gender: '',
}

export default FriendListItem
