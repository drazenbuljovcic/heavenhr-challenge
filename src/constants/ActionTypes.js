export const ADD_FRIEND = 'ADD_FRIEND';
export const STAR_FRIEND = 'STAR_FRIEND';
export const DELETE_FRIEND = 'DELETE_FRIEND';

export const REFRESH_PAGINATION = 'REFRESH_PAGINATION';
export const CHANGE_PAGINATION_PAGE = 'CHANGE_PAGINATION_PAGE';