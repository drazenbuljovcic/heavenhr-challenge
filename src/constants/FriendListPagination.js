export const ITEM_COUNT = 2;