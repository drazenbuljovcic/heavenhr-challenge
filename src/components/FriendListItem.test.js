import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import toJson from 'enzyme-to-json';

import * as genderTypes from '../constants/GenderTypes';

import FriendListItem from './FriendListItem';

configure({ 'adapter': new Adapter() });

describe('A Friend List Item', () => {
  const defaultProps = {
    key: 0,
    id: 0,
    name: '',
    gender: '',
    starred: false,
    starFriend: jest.fn(),
    deleteFriend: jest.fn(),
  };
  const component = shallow(<FriendListItem {...defaultProps} />);

  it('should match it\'s empty snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should display the friend name', () => {
    expect(toJson(shallow(
      <FriendListItem {...defaultProps} name='Friend' />
    ))).toMatchSnapshot();
  });

  it('should display gender icon if gender is specified', () => {
    expect(toJson(shallow(
      <FriendListItem {...defaultProps} name='Friend' gender={genderTypes.MALE}/>
    ))).toMatchSnapshot();
  });

  it('should be able to star the friend', () => {
    component.find('button[aria-label=\'Star Friend\']').simulate('click');
    expect(defaultProps.starFriend).toHaveBeenCalled();
  });

  it('should be able to delete the friend', () => {
    component.find('button[aria-label=\'Delete Friend\']').simulate('click');
    expect(defaultProps.deleteFriend).toHaveBeenCalled();
  });
});