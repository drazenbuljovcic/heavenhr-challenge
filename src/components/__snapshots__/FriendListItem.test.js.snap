exports[`A Friend List Item should display gender icon if gender is specified 1`] = `
<li>
  <div>
    <div>
      <span>
        Friend
      </span>
      <i
        aria-hidden="true"
        className="fa fa-male" />
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div>
    <button
      aria-label="Star Friend"
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-star-o" />
    </button>
    <button
      aria-label="Delete Friend"
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-trash" />
    </button>
  </div>
</li>
`;

exports[`A Friend List Item should display the friend name 1`] = `
<li>
  <div>
    <div>
      <span>
        Friend
      </span>
      <i
        aria-hidden="true"
        className="fa" />
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div>
    <button
      aria-label="Star Friend"
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-star-o" />
    </button>
    <button
      aria-label="Delete Friend"
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-trash" />
    </button>
  </div>
</li>
`;

exports[`A Friend List Item should match it's empty snapshot 1`] = `
<li>
  <div>
    <div>
      <span />
      <i
        aria-hidden="true"
        className="fa" />
    </div>
    <div>
      <small>
        xx friends in common
      </small>
    </div>
  </div>
  <div>
    <button
      aria-label="Star Friend"
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-star-o" />
    </button>
    <button
      aria-label="Delete Friend"
      className="btn btn-default undefined"
      onClick={[Function]}>
      <i
        className="fa fa-trash" />
    </button>
  </div>
</li>
`;
